File: go/vertical_slide/features/create_account/repository.go
package create_account

import (
	"github.com/scylladb/gocqlx/v2/qb"
	scyna "github.com/scyna/core"
)

const TABLE_NAME = "ex_account.account"

type repository struct {
	context *scyna.Context
}

func LoadRepository(context *scyna.Context) *repository {
	return &repository{context: context}
}

func (r *repository) GetAccount(email string) (*Account, scyna.Error) {
	var ret Account
	if err := qb.Select(TABLE_NAME).
		Columns("id", "name", "email").
		Where(qb.Eq("email")).
		Query(scyna.DB).
		Bind(email).
		GetRelease(&ret); err != nil {
		r.context.Error(err.Error())
		return nil, ACCOUNT_NOT_EXISTED
	}
	return &ret, nil
}

func (r *repository) CreateAccount(account *Account) scyna.Error {

	if err := qb.Insert(TABLE_NAME).
		Columns("id", "name", "email", "password").
		Query(scyna.DB).
		BindStruct(account).
		ExecRelease(); err != nil {
		r.context.Error(err.Error())
		return scyna.SERVER_ERROR
	}
	return nil
}

File: csharp/account/repository/Repository.cs
namespace ex.account;

using scyna;
using Scylla.Net;

public class Repository : IRepository
{
    const string TABLE_NAME = "account";
    const string KEYSPACE = "ex_account";
    private Context context;

    private Repository(Context context) { this.context = context; }

    public static IRepository load(Context context)
    {
        return new Repository(context);
    }

    public void createAccount(Command command, Account account)
    {
        if (account.Email == null || account.Name == null) throw scyna.Error.BAD_DATA;
        var insert = Engine.DB.Session.Prepare(string.Format("INSERT INTO {0}.{1}(id,email,name) VALUES(?,?,?)", KEYSPACE, TABLE_NAME));
        command.Batch.Add(insert.Bind((long)account.ID, account.Email.ToString(), account.Name.ToString()));
    }

    public Account GetAccountByEmail(EmailAddress email)
    {
        var query = Engine.DB.Session.Prepare(string.Format("SELECT id,email,name FROM {0}.{1} WHERE email=? LIMIT 1", KEYSPACE, TABLE_NAME));
        var statement = query.Bind(email.ToString());
        return queryAccount(statement);
    }

    public Account GetAccountByID(ulong ID)
    {
        var query = Engine.DB.Session.Prepare(string.Format("SELECT id,email,name FROM {0}.{1} WHERE id=? LIMIT 1", KEYSPACE, TABLE_NAME));
        var statement = query.Bind((long)ID);
        return queryAccount(statement);
    }

    private Account queryAccount(BoundStatement statement)
    {
        try
        {
            var rs = Engine.DB.Session.Execute(statement);
            var row = rs.First();
            var account = new Account(context);
            account.ID = (ulong)row.GetValue<long>("id");
            account.Email = EmailAddress.Parse(row.GetValue<string>("email"));
            account.Name = Name.Create(row.GetValue<string>("name"));
            return account;
        }
        catch (InvalidOperationException)
        {
            throw Error.ACCOUNT_NOT_FOUND;
        }
        catch
        {
            throw scyna.Error.SERVER_ERROR;
        }
    }
}